packages/flutter/lib/src/material/chip.dart

```diff
@@ -245,6 +245,38 @@ abstract class DeletableChipAttributes {
   String get deleteButtonTooltipMessage;
 }
 
+/// An interface for material design chips that can have check marks.
+///
+/// The defaults mentioned in the documentation for each attribute are what
+/// the implementing classes typically use for defaults (but this class doesn't
+/// provide or enforce them).
+///
+/// See also:
+///
+///  * [InputChip], a chip that represents a complex piece of information, such
+///    as an entity (person, place, or thing) or conversational text, in a
+///    compact form.
+///  * [FilterChip], uses tags or descriptive words as a way to filter content.
+///  * <https://material.io/design/components/chips.html>
+abstract class CheckmarkableChipAttributes {
+  // This class is intended to be used as an interface, and should not be
+  // extended directly.
+  factory CheckmarkableChipAttributes._() => null;
+
+  /// Whether or not to show a check mark when [selected] is true.
+  ///
+  /// Defaults to true.
+  final bool showCheckmark;
```

Comment:
To stick with the pattern, this should probably be a getter.

```suggestion
  bool get showCheckmark;
```

Reply:
Done